ush/merge_fv3_chem_tile.py

```diff
@@ -40,36 +43,91 @@
+		sys.exit(104)
+
+	if not os.path.isfile(rest_file_name):
+		print("FATAL ERROR: Atmosphere restart file " + rest_file_name + " does not exist!")
+		sys.exit(105)
+
 	if not os.path.isfile(append_file_name):
 		print("FATAL ERROR: Chemistry file " + append_file_name + " does not exist!")
-		sys.exit(103)
+		sys.exit(106)
 
 	append_file = netCDF4.Dataset(append_file_name, "r")
 	base_file = netCDF4.Dataset(base_file_name, "r+")
+	core_file = netCDF4.Dataset(core_file_name, "r")
+	ctrl_file = netCDF4.Dataset(ctrl_file_name, "r")
+	rest_file = netCDF4.Dataset(rest_file_name, "r")
+
+	# read pressure layer thickness from restart file
+	delp = rest_file["delp"][0,:]
+	# read a, b coefficients to generate sigma levels
+	ak = core_file["ak"][0,:]
+	bk = core_file["bk"][0,:]
+
+	# read surface pressure from initial conditions file
+	psfc = base_file["ps"][:,:]
+	# read sigma-level a, b coefficients from initial conditions control file
+	ai = ctrl_file["vcoord"][0,1:]
+	bi = ctrl_file["vcoord"][1,1:]
+
+	# IC sigma levels must match model restart sigma levels
+	if ak.size != ai.size:
+		print("FATAL ERROR: Inconsistent size of A(k) arrays: src=",ak.size,", dst=",ai.size)
+		sys.exit(107)
+
+	if bk.size != bi.size:
+		print("FATAL ERROR: Inconsistent size of B(k) arrays: src=",bk.size,", dst=",bi.size)
+		sys.exit(108)
+
+	if not np.array_equal(ak,ai):
+		print("FATAL ERROR: A(k) coefficients must be identical")
+		sys.exit(109)
+
+	if not np.array_equal(bk,bi):
+		print("FATAL ERROR: B(k) coefficients must be identical")
+		sys.exit(110)
+
+	dp = np.zeros(delp.shape)
+	for k in range(0, dp.shape[0]):
+		dp[k,:,:] = ak[k+1]-ak[k] + psfc * (bk[k+1]-bk[k])
+
+	scale_factor = delp / dp
 
 	# print(base_file)
 	# print(base_file.dimensions["ntracer"])
 
 	old_ntracer = base_file.dimensions["ntracer"].size
 	new_ntracer = old_ntracer + len(tracers_to_append)
 
-	# Copy over chemistry dimensions
-	for dim_name in append_file.dimensions:
-		base_file.createDimension(dim_name, append_file.dimensions[dim_name].size)
+	print("Adding the following variables to " + base_file_name + ":\n")
 
+	print(" Name   | Total mass (restart) | Total mass (IC)      | Max column abs. diff.")
+	print("-"*8 + "+" + "-"*22 + "+" + "-"*22 + "+" + "-"*24)
 	for variable_name in tracers_to_append:
-		print("Adding variable " + variable_name + " to file")
 		variable = append_file[variable_name]
-		base_file.createVariable(variable_name, variable.datatype, variable.dimensions)
-		base_file[variable_name][:] = variable[:]
+		base_file.createVariable(variable_name, variable.datatype, base_file["sphum"].dimensions)
+		base_file[variable_name][0,:,:]  = 0.
+		base_file[variable_name][1:,:,:] = scale_factor * variable[0,:,:,:]
 		base_file[variable_name].setncatts(variable.__dict__)
+		msrc = variable * delp
+		mdst = base_file[variable_name][1:,:,:] * dp
+		print(' {0:6}   {1:20}   {2:20}    {3:22}'.format(variable_name, np.sum(msrc), np.sum(mdst), np.max(np.abs(msrc-mdst))))
```

Comment:
- Should this print only be done for debug?
- This string would be better executed using an f-string. At the very least, use named fields so the string is readable.

Reply:
I would prefer to leave this output enabled by default since it could help debug mass-conservation issues down the line.

Commit 8b020e0c36c1f454fa36a8cd26e04a38ad295408 implements your suggestion (f-string) and adds self-explanatory variable names.